lib/health_check.rb

```diff
@@ -0,0 +1,91 @@
+class HealthCheck
+  include ChefResource
+
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  KEYERROR = 'key error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = nil
```

Comment:
I believe you that it wasn't arbitrary, but based on what you just said I don't see why it should therefore be initialized to nil. Yes, the overall method ensures it isn't nil at this time, but it seems wise to avoid having a nil value from the start if we can to avoid future issues around if this code is modified and someone messes with the status outside of the overall method. Why even have the cognitive overhead of having to think of status as nil, especially given that we've already defaulted to assuming the services are working, that would imply we should default to assuming status is ok.